src/server/routes/routes.ts

```diff
@@ -0,0 +1,189 @@
+import { processOrientationOption } from '../../browser/helpers';
+import generatePdf, { previewPdf } from '../../browser';
+import { SendingFailedError, PDFNotFoundError } from '../errors';
+import config from '../../common/config';
+import { OPTIONS_HEADER_NAME } from '../../common/consts';
+import { PreviewReqQuery } from '../../common/types';
+
+const PORT = config.webPort;
+
+type PreviewOptions = unknown;
+type PreviewHandlerRequest = Request<
+  PreviewOptions,
+  any,
+  unknown,
+  PreviewReqQuery
+>;
+
+export type GenerateHandlerReqyest = Request<
+  unknown,
+  unknown,
+  Record<string, any>,
+  { service: ServiceNames; template: string }
+>;
+
+export type HelloHandlerRequest = Request<
+  unknown,
+  unknown,
+  unknown,
+  { policyId: string; totalHostCount: number }
+>;
+
+export interface PupetterBrowserRequest
+  extends Request<
+    unknown,
+    unknown,
+    unknown,
+    { service: ServiceNames; template: string }
+  > {
+  headers: IncomingHttpHeaders & {
+    [OPTIONS_HEADER_NAME]?: string;
+  };
+}
+
+const APIPrefix = '/api/crc-pdf-generator/v1';
+
+const router = Router();
+
+// Middlware that activates on all routes, responsible for rendering the correct
+// template/component into html to the requester.
+router.use('^/$', async (req: PupetterBrowserRequest, res, _next) => {
+  let service: ServiceNames = req.query.service;
+  let template: string = req.query.template;
+  if (!service) {
+    console.info('Missing service, using "demo"');
+    service = ServiceNames.demo;
+  }
+  if (!template) {
+    console.info('Missing template, using "demo"');
+    template = 'demo';
+  }
+
+  const templateConfig = {
+    service,
+    template,
+  };
+  try {
+    const configHeaders: string | undefined = req.headers[OPTIONS_HEADER_NAME];
+    if (configHeaders) {
+      delete req.headers[OPTIONS_HEADER_NAME];
+    }
+
+    const templateData = await getTemplateData(
+      req.headers,
+      templateConfig,
+      configHeaders ? JSON.parse(configHeaders) : undefined
+    );
+    const HTMLTemplate: string = renderTemplate(
+      templateConfig,
+      templateData as Record<string, unknown>
+    );
+    res.send(HTMLTemplate);
+  } catch (error) {
+    console.log(error);
+    res.send(`<div>Unable to render ${template}!</div>`);
+  }
+});
+
+router.get(`${APIPrefix}/hello`, (_req, res) => {
+  return res.status(200).send('<h1>Well this works!</h1>');
+});
+
+router.post(
+  `${APIPrefix}/generate`,
+  async (req: GenerateHandlerReqyest, res) => {
+    const rhIdentity = req.headers['x-rh-identity'] as string;
+    const orientationOption = processOrientationOption(req);
+
+    if (!rhIdentity) {
+      return res.status(401).send('Unauthorized access not allowed');
+    }
+
+    const service = req.body.service;
+    const template = req.body.template;
+    const dataOptions = req.body;
+
+    const tenant = JSON.parse(atob(rhIdentity))['identity']['internal'][
+      'org_id'
+    ];
```

Comment:
Same for this, isn't there a dependency which would parse this and passed it to each method?

Reply:
This is actually useless value. Does nothing just logs the header. And we can use header for that.